File: backend/src/main/java/com/dev/backend/constant/VoucherStatus.java
package com.dev.backend.constant;

import com.dev.backend.exception.BadRequestException;

public enum VoucherStatus {
    ACTIVE,
    INACTIVE,
    DELETED;

    public static VoucherStatus from(String status) {
        if (status == null || status.isBlank()) {
            return null;
        }

        try {
            return valueOf(status.trim().toUpperCase());
        } catch (IllegalArgumentException e) {
            throw new BadRequestException("Invalid voucher status.");
        }
    }
}

File: frontend/src/modules/admin/voucher/types/voucher.status.ts
export const VoucherStatus = {
  ACTIVE: "ACTIVE",
  INACTIVE: "INACTIVE",
  DELETED: "DELETED",
} as const;

export type VoucherStatus = (typeof VoucherStatus)[keyof typeof VoucherStatus];

export const VoucherStatusConfig: Record<VoucherStatus, { label: string }> = {
  ACTIVE: { label: "Hoạt động" },
  INACTIVE: { label: "Tạm ngưng" },
  DELETED: { label: "Đã xóa" },
};

export const getVoucherStatusLabel = (status?: VoucherStatus | string) => {
  if (!status) return "Không xác định";
  const upperStatus = String(status).toUpperCase() as VoucherStatus;
  return VoucherStatusConfig[upperStatus]?.label || String(status);
};
